README.md

```diff
@@ -70,89 +84,148 @@ git push origin <add-your-branch-name>
+Address the comments of the review or reply to them explaining your reasoning. If see that the suggested changes make sense, make the changes locally and save. Then add, commit and push the updates to GitHub as you did with the initial changes.
+
+Use a meaningful commit message when addressing the comments eg. `Improve wording` instead of `addressing comments`
 
 ### Where to go from here?
 
 Congrats!  You just completed the standard clone -> edit -> PR_ workflow that helps you collaborate and share knowledge with your team!
 
-## Part II - Rewrite history
 
-Rewritting history in git is possible. One should be careful not to do that, in any branch that is already shared with somebody else. To be on the safe side do history rewrites before pushing the commits or in another branch.
+## Part II - Undoing changes
+
+If you realize that you want to undo some changes you've made, you have several options and it depends on what you have done before.
+Rewritting history in git is possible. One should be careful not to do that, in general because changes might get lost and in particular on any branch that is already shared with somebody else. 
+To be on the safe side do history rewrites **before** pushing the commits or even better in an additional branch!
 
+### Your options
 
-### Get the ugly-history
+- If you have not added (we call it "staged") or commited a change yet `git checkout <filename>` removes all of the local changes on the file.  Be careful with this one, you might easily lose relevant work
+- If you have already added but not yet commited a change `git reset HEAD <filename>`
 
-Start a branch from the `ugly-history` branch
+For changes that have already been commited:
+
+- `git commit --amend -m "your new commit message"` lets you add some changes to your previous commit (use on private branches only)
+- `git revert <commitID>` lets you undo a specific commit as an additional commit. This is a very safe option also if you shared a branch already
+- `git reset --soft <commitID>` there is also a `--hard` flag but don't use it unless you know exactly what you are doing!
+
+
+### Rewriting history
+
+Get the ugly-history branch and start a new branch from it
 
-Tip
 ```
-git checkout <BRANCH_TO_BASE_YOURS>
-git checkout -b <NEW_BRANCH>
+git checkout <ugly-history>
+git checkout -b <your_new_branch>
 ``` 
 
-## Look at the commits
+### Look at the commits
+
+Use git log to look at the commit history (last 10 commits)
+
+```
+git log -10
+```
 
-Use git log to look at the commit messages
+### Undo a commit
 
+Revert the commit where `commit-message.md` is added by running
 ```
-git log -5
+git revert --no-edit <commit hash> 
 ```
+and again check your commit history. The commit ID can be obtained from the commit history.
+
 
-Optional things
+### Rewrite the last 5 commits
 
-Tip:
-to make your git log look more compact like this:
```

Comment:
You are removing the tip with gitconfig. Do you think it was too confusing or not working?